Modules/_datetimemodule.c

```diff
@@ -1990,69 +1948,68 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
                     goto Done;
                 }
             }
-            ntoappend = PyOS_snprintf(buf, sizeof(buf),
+            /* Buffer of maximum size of formatted year permitted by long.
+             * +6 to accommodate dashes, 2-digit month and day for %F. */
+            char buf[SIZEOF_LONG * 5 / 2 + 2 + 6];
```

Comment:
It may be interesting to add a test on LONG_MAX year to check for buffer overflow.

Reply:
Right. I saw later that this code exists before your PR.